operators/src/upgrade/helm/client.rs

```diff
@@ -9,62 +9,117 @@ use crate::upgrade::common::error::Error;
 #[derive(Debug, Clone, Default)]
 struct HelmArgs {
-    name: Option<String>,
+    name: String,
     opts: Vec<(String, String)>,
     namespace: Option<String>,
     values: Vec<PathBuf>,
 }
 
 impl HelmArgs {
     /// Set a name.
-    fn set_name(&mut self, name: String) {
-        self.name = Some(name);
+    fn with_name(mut self, name: String) -> Self {
+        self.name = name;
+        self
     }
 
     /// Set a single option.
-    fn set_opt(&mut self, key: String, value: String) {
+    fn with_opt(mut self, key: String, value: String) -> Self {
         self.opts.push((key, value));
+        self
     }
 
     /// Reset array of options.
-    fn set_opts(&mut self, options: Vec<(String, String)>) {
+    fn with_opts(mut self, options: Vec<(String, String)>) -> Self {
         self.opts = options;
+        self
     }
 
     /// Set namepsace.
-    fn set_namespace(&mut self, ns: String) {
+    fn with_namespace(mut self, ns: String) -> Self {
         self.namespace = Some(ns);
+        self
     }
 
-    /// Set one value.
-    fn set_values(&mut self, values: Vec<PathBuf>) {
+    /// Set values.
+    fn with_values(mut self, values: Vec<PathBuf>) -> Self {
         self.values = values;
+        self
     }
 
     /// Set one value.
-    fn set_value(&mut self, value: PathBuf) {
+    fn with_value(mut self, value: PathBuf) -> Self {
         self.values.push(value);
+        self
     }
 
-    /// Set a name.
-    fn get_name(&self) -> &String {
-        self.name.as_ref().unwrap()
+    /// Get a name.
+    fn name(&self) -> &String {
+        &self.name
     }
 
-    /// Set a single option.
-    fn get_opts(&self) -> &Vec<(String, String)> {
+    /// Get options.
+    fn opts(&self) -> &Vec<(String, String)> {
         &self.opts
     }
 
-    /// Set namepsace.
-    fn get_namespace(&self) -> &String {
+    /// Get namespace.
+    fn namespace(&self) -> &String {
         self.namespace.as_ref().unwrap()
     }
 
-    /// Set one value.
-    fn get_values(&self) -> &Vec<PathBuf> {
+    /// Get values.
+    fn values(&self) -> &Vec<PathBuf> {
         &self.values
     }
+
+    // Apply arguments for helm command.
+    fn apply_args(&self, command: &mut Command) {
+        command
+            .arg(self.name())
+            .arg("--namespace")
+            .arg(self.namespace());
+
+        for value_path in self.values() {
+            command.arg("--values").arg(value_path);
+        }
+
+        for (key, val) in self.opts() {
+            command.arg("--set").arg(format!("{}={}", key, val));
+        }
+    }
+
+    // Run upgrade command.
+    fn upgrade(self) -> Result<Output, Error> {
+        let mut cmd = self.clone().command(vec!["upgrade".to_string()]);
+        self.run(&mut cmd)
+    }
+
+    // Run get values command.
+    fn get_values(self) -> Result<Output, Error> {
+        let mut cmd = self
+            .clone()
+            .command(vec!["get".to_string(), "values".to_string()]);
+        self.run(&mut cmd)
```

Comment:
do we need to create a command at all?
Why not simply:
```rust
fn run(self, command_type_args: Vec<String>) -> {
  let mut command = Command::new("helm");
  for c in command_type_args {
      command.arg(c);
  }
  self.apply_args(&mut command);
  command.arg("--output=json");
  command.output()
      .map_err(|_| Error::HelmStd("Error while running helm get command".to_string()))
}
...
self.run(vec!["get".to_string(), "values".to_string()])

```

Reply:
Btw, since you're using `Command` why not implementing it like they do, and instead of using Vec<String> which required to do `vec!["get".to_string(), "values".to_string()]`, use traits so you can pass &str instead: https://doc.rust-lang.org/std/process/struct.Command.html#method.args
So it could become:
```rust
self.run(["get", "values"])
```